src/Store.js

```diff
@@ -219,26 +220,31 @@ class Store {
     amount = amount || this.options.maxHistory
     fetchEntryTimeout = fetchEntryTimeout || this.options.fetchEntryTimeout
 
-    const localHeads = await this._cache.get(this.localHeadsPath) || []
-    const remoteHeads = await this._cache.get(this.remoteHeadsPath) || []
-    const heads = localHeads.concat(remoteHeads)
+    try {
+      await this._cache.open()
+      const localHeads = await this._cache.get(this.localHeadsPath) || []
+      const remoteHeads = await this._cache.get(this.remoteHeadsPath) || []
+      const heads = localHeads.concat(remoteHeads)
 
-    if (heads.length > 0) {
-      this.events.emit('load', this.address.toString(), heads)
-    }
+      if (heads.length > 0) {
+        this.events.emit('load', this.address.toString(), heads)
+      }
+
+      await mapSeries(heads, async (head) => {
+        this._recalculateReplicationMax(head.clock.time)
+        const log = await Log.fromEntryHash(this._ipfs, this.identity, head.hash, { logId: this._oplog.id, access: this.access, sortFn: this.options.sortFn, length: amount, exclude: this._oplog.values, onProgressCallback: this._onLoadProgress.bind(this), timeout: fetchEntryTimeout })
+        await this._oplog.join(log, amount)
+      })
 
-    await mapSeries(heads, async (head) => {
-      this._recalculateReplicationMax(head.clock.time)
-      const log = await Log.fromEntryHash(this._ipfs, this.identity, head.hash, { logId: this._oplog.id, access: this.access, sortFn: this.options.sortFn, length: amount, exclude: this._oplog.values, onProgressCallback: this._onLoadProgress.bind(this), timeout: fetchEntryTimeout })
-      await this._oplog.join(log, amount)
-    })
+      // Update the index
+      if (heads.length > 0) {
+        await this._updateIndex()
+      }
 
-    // Update the index
-    if (heads.length > 0) {
-      await this._updateIndex()
+      this.events.emit('ready', this.address.toString(), this._oplog.heads)
+    } catch (e) {
+      console.warn('Warning: Database closed while loading - please consider `await db.load()`')
```

Comment:
Is the only condition an error will be thrown that `await` isnt used when calling `load`?

Reply:
It's a high probability that it's the only thing that will throw an exception here in practice, but we should still at least throw a warning for generic errors. I still think we should fully bail here but keep catching the error and displaying a warming.

This is the test this fixes in `orbit-db`:

https://github.com/orbitdb/orbit-db/blob/master/test/persistency.js#L123